### ISSUE TITLE
base @mod return type on denominator value when comptime-known

### ISSUE THREAD

**Issue by marler8997** [ziglang/zig#9000]
The return type of `@mod` could be limited when the denominator is known at comptime. 
```zig
@mod(numerator: T, denominator: T) std.math.IntFittingRange(0, denominator-1)
```

As an extension to this, the `@div*` functions could also use a similar technique to limit the return type.

**Comment by andrewrk** [comment]
This looks like something already addressed by the accepted proposal #509.

**Comment by marler8997** [comment]
Yes it looks like with that proposal we should be good to go, closing in favor of #509
